src/main/java/com/google/sps/data/FakeScheduledInterviewDao.java

```diff
@@ -0,0 +1,106 @@
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Random;
+import java.util.Set;
+
+/** Accesses Datastore to support managing ScheduledInterview entities. */
+public class FakeScheduledInterviewDao implements ScheduledInterviewDao {
+  // @param datastore The DatastoreService we're using to interact with Datastore.
+  public Map<String, ScheduledInterview> datastore;
+
+  /** Initializes the fields for ScheduledInterviewDatastoreDAO. */
+  public FakeScheduledInterviewDao() {
+    datastore = new LinkedHashMap<String, ScheduledInterview>();
+  }
+
+  /**
+   * Retrieves a scheduledInterviewEntity from Datastore and returns it as a ScheduledInterview
+   * object.
+   */
+  @Override
+  public Optional<ScheduledInterview> get(long id) {
+    if (datastore.containsKey(Long.toString(id))) {
+      return Optional.of(datastore.get(Long.toString(id)));
+    }
+    return Optional.empty();
+  }
+
+  /**
+   * Retrieves all scheduledInterview entities from Datastore that involve a particular user and
+   * returns them as a list of ScheduledInterview objects in the order in which they occur.
+   */
+  @Override
+  public List<ScheduledInterview> getForPerson(String email) {
+    List<ScheduledInterview> relevantInterviews = new ArrayList<>();
+    List<ScheduledInterview> scheduledInterviews =
+        new ArrayList<ScheduledInterview>(datastore.values());
+    scheduledInterviews.sort(
+        (ScheduledInterview s1, ScheduledInterview s2) ->
+            s1.when().start().isBefore(s2.when().start()) ? -1 : 0);
+    for (ScheduledInterview scheduledInterview : scheduledInterviews) {
+      if (email.equals(scheduledInterview.interviewerEmail())
+          || email.equals(scheduledInterview.intervieweeEmail())) {
+        relevantInterviews.add(scheduledInterview);
+      }
+    }
+    return relevantInterviews;
+  }
+
+  /** Creates a ScheduledInterview Entity and stores it in Datastore. */
+  @Override
+  public void create(ScheduledInterview scheduledInterview) {
+    long generatedId = new Random().nextLong();
```

Comment:
How do you know this generatedId won't collide with other id's in the past? Need to make sure id is not in the hashmap, else keep generating a new one until it doesn't.

Reply:
In practice usually tests just assume that the id wont collide. The chance of a collision is very very small provided that the values are randomly distributed and the key space is large.

Reply:
Oh, my bad, this isn't a test.